### ISSUE TITLE
Launching singularity jobs on gpu-enabled mesos agents

### ISSUE THREAD

**Issue by filip-andrei** [HubSpot/Singularity#1346]
As far as i read, mesos currently requires frameworks to be flagged as supporting gpu resources to be able to receive offers from agents that are gpu aware. Is there currently some option to flag singularity as such ?

**Comment by ssalinas** [comment]
Right now Singularity does not have support for gpu resources, we would need to make changes to add that

**Comment by kchaliki** [comment]
Hello, I had a quick go at this and came up with the following. My tests don't pass yet but is this the right kind of approach? Also I think GPU support is added on mesos 1.0, is that a blocker for Singularity? 

https://github.com/kchaliki/Singularity/commit/d1edfe6bafcaa2b990a44d8d277081d3dfd95c7a

**Comment by ssalinas** [comment]
@kchaliki , correct, d1edfe6 is on the right track. Happy to review the PR if you want to finish that off.

As for the mesos side, it is on our roadmap for this year to upgrade to mesos 1.x. Not sure of an exact timeline yet however

**Comment by kchaliki** [comment]
Thanks, I'll send a pull request when done

**Comment by yuriy-filatov** [comment]
Hey guys. Any updates on GPU support in singularity?
